pkg/asset/cluster/tfvars/tfvars.go

```diff
@@ -843,6 +843,14 @@ func (t *TerraformVariables) Generate(ctx context.Context, parents asset.Parents
 		}
 		switch installConfig.Config.Publish {
 		case types.InternalPublishingStrategy:
+			if installConfig.Config.PowerVS != nil {
+				metadata, err := powervsconfig.NewMetadata(installConfig.Config)
+				if err != nil {
+					return err
+				}
+				installConfig.PowerVS = metadata
+			}
+
```

Comment:
Same here.

Reply:
I don't know why I got into this Terraform area when running with CAPI, but I had to do this to get further. I can retest this, as this is one of the earlier changes I put in place, before the whole refactor effort, and since you are asking for 2 separate commits, I think I have to start over and scrutinize one by one.

Reply:
No longer part of this PR.